tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```diff
@@ -73,28 +82,79 @@ spec:
             echo "No valid file was provided as signRegistryAccessPath."
             exit 1
         fi
+        PUBLIC_KEY_FILE=$(mktemp)
+        echo -n "$PUBLIC_KEY" > "$PUBLIC_KEY_FILE"
+        RUNNING_JOBS="\j" # Bash parameter for number of jobs currently running
 
+        jobpid(){
+            pid=$(cut -d' ' -f4 < /proc/self/stat)
+            echo "$pid"
+        }
+        echopid(){
+            pid=$(jobpid)
+            echo "${pid}: $*"
+        }
         run_cosign () { # Expected arguments are [digest_reference, tag_reference]
-            # Upload transparency log when rekor url is specified
-            if [ -v REKOR_URL ]; then
-                COSIGN_COMMON_ARGS="-y --rekor-url=$REKOR_URL --key $SIGN_KEY"
-            else
-                COSIGN_COMMON_ARGS="--tlog-upload=false --key $SIGN_KEY"
-            fi
-            echo "Signing manifest $1 ($2)"
             attempt=0
+            backoff1=2
+            backoff2=3
             until [ "$attempt" -gt "$(params.retries)" ] ; do # 3 retries by default
-                cosign -t 3m0s sign\
-                ${COSIGN_COMMON_ARGS}\
-                --sign-container-identity "$2"\
-                "$1" && break
+                cosign "$@" && break
```

Comment:
@mmalina we were just talking about a util for backoff retries in refinement today. This looks like code we might want to copy to utils (please resolve this once seeing it)

Reply:
Yeah, possibly.